tests/engines/xarray_engine/test_plugin.py

```diff
@@ -2,211 +2,141 @@
 # Licensed under the MIT License.
 
 import numpy as np
+
+import xarray as xr
+from xarray import Variable, open_dataset
+from xarray.testing import assert_equal
+
 import pytest
+from pytest import importorskip
+
+tiledb = importorskip("tiledb")
+
+INT_1D_DATA = np.arange(0, 16, dtype=np.int32)
+INT_1D_ATTRS = {"description": "int 1D data var"}
+INT_1D_VAR = Variable(["rows"], INT_1D_DATA, INT_1D_ATTRS)
+INT_1D_COORDS = {"rows": np.arange(0, 16, dtype=np.int32)}
+
+INT_2D_DATA = np.arange(0, 32, dtype=np.int32).reshape(8, 4)
+INT_2D_ATTRS = {"description": "int 2D data var"}
+INT_2D_VAR = Variable(["rows", "cols"], INT_2D_DATA, INT_2D_ATTRS)
+INT_2D_COORDS = {"rows": np.arange(0, 8, dtype=np.int32), "cols": np.arange(0, 4, dtype=np.int32)}
+
+INT_1D_SHIFTED_ATTRS = {"description": "int 1d shifted data var"}
+INT_1D_SHIFTED_VAR = Variable(["rows"], INT_1D_DATA, INT_1D_SHIFTED_ATTRS)
+INT_1D_SHIFTED_COORDS = {"rows": np.arange(5, 21, dtype=np.int32)}
+
+INT_2D_SHIFTED_ATTRS = {"description": "int 2d shifted data var"}
+INT_2D_SHIFTED_VAR = Variable(["rows", "cols"], INT_2D_DATA, INT_2D_SHIFTED_ATTRS)
+INT_2D_SHIFTED_COORDS = {"rows": np.arange(4, 12, dtype=np.int32), "cols": np.arange(1, 5, dtype=np.int32)}
+
+DATETIME_1D_VAR_ATTRS = {"description": "datetime 1d data var"}
+DATETIME_1D_VAR = Variable(["rows"], INT_1D_DATA, DATETIME_1D_VAR_ATTRS)
+DATETIME_1D_COORDS = {"rows": np.arange(np.datetime64("2000-01-01"), np.datetime64("2000-01-17"), np.timedelta64(1, 'D'))}
+
+DS_ATTRS = {"description": "test dataset"}
+DATETIME_1D_DS_ATTRS = {**DS_ATTRS, "rows": {"time reference": np.datetime64("2000-01-01"), "freq": "D"}}
+
+SAMPLE_DATASETS = {
+    "simple_1d": xr.Dataset(data_vars={"data": INT_1D_VAR}, coords=INT_1D_COORDS, attrs=DS_ATTRS),
+    "simple_2d": xr.Dataset(data_vars={"data": INT_2D_VAR}, coords=INT_2D_COORDS, attrs=DS_ATTRS),
+    "1d_shifted": xr.Dataset(data_vars={"data": INT_1D_SHIFTED_VAR}, coords=INT_1D_SHIFTED_COORDS, attrs=DS_ATTRS),
+    "2d_shifted": xr.Dataset(data_vars={"data": INT_2D_SHIFTED_VAR}, coords=INT_2D_SHIFTED_COORDS, attrs=DS_ATTRS),
+    "1d_datetime": xr.Dataset(data_vars={"data": DATETIME_1D_VAR}, coords=DATETIME_1D_COORDS, attrs=DATETIME_1D_DS_ATTRS)
+}
+
+class TestTDBBackend:
+    engine = "tiledb"
+
+    def open(self, path):
+        with open_dataset(path, engine=self.engine) as ds:
+            ds_obj = ds
+        return ds_obj
+
+    def to_tiledb(self, dataset, path):
```

Comment:
```suggestion
    def to_tiledb(self, dataset, path):  # noqa: C901
```
This will allow flake8 to pass. C901 is a warning that the function is "too complex" based on how many if/else statements and for loops there are in the funciton.